feat(service-catalog): Terraform support for integrations and services (#2586)
Kong/developer.konghq.com: app/service-catalog/services.md

---
title: "Service Catalog services"
content_type: reference
layout: reference

products:

>     - gateway
>     - service-catalog

Reviewer: ```suggestion
    - service-catalog
    - gateway
```
The first one is what gets displayed in the sidebar. We should probably fix that, and display all of them if possible 🤔 

Editing to add, because now I'm not sure: Are Service Catalog services at all related to Gateway? Does `gateway` even need to be here?

Author: @lena-larionova Good call out. Looks like for the other Konnect app reference pages we only use the app, so I'll remove `gateway`.

works_on:
  - konnect

description: Learn about services in Service Catalog and how to configure them.

breadcrumbs:
  - /service-catalog/

related_resources:
  - text: "Service Catalog"
    url: /service-catalog/
  - text: Scorecards
    url: /service-catalog/scorecards/
  - text: Traceable integration
    url: /service-catalog/integrations/traceable/
  - text: GitHub integration
    url: /service-catalog/integrations/github/
  - text: GitLab integration
    url: /service-catalog/integrations/gitlab/
  - text: SwaggerHub integration
    url: /service-catalog/integrations/swaggerhub/
  - text: Datadog integration
    url: /service-catalog/integrations/datadog/
  - text: PagerDuty integration
    url: /service-catalog/integrations/pagerduty/

> ---

Reviewer: I think it would be helpful to add an FAQ here that says something like "What's the different between a Gateway Service and a Service Catalog service?". We even have definitions for both in our glossary: https://developer.konghq.com/glossary/

> A Service Catalog service can be thought of as a collection of one or more resources because you can associate resources from one or more integrations with them.

Author: ```suggestion
A Service Catalog service is a collection of one or more resources from integrations.
```

A Service Catalog service represents the following:
* A unit of software that is typically owned by a single team
* Exposes one or more APIs 
* May be dependent on other Service Catalog services (as either upstream or downstream)  

Service Catalog services allow you to gain visibility into all resources in your organization, including what teams or people own them. 

To create a Service Catalog service, do one of the following:

{% navtabs "service" %}
{% navtab "UI" %}

> In the {{site.konnect_short_name}} UI, click **Service Catalog** > **Services** in the sidebar. Click **New service** and configure the details about your service. Then, you can map a service to an integration's resource by clicking on the service and then select "Map resources" from the **Action** dropdown menu.

Reviewer: ```suggestion
1. In the {{site.konnect_short_name}} UI, navigate to **Service Catalog** > **Services** in the sidebar. 
1. Click **New service** and configure the details about your service. 
1. Map the service to an integration: click the service, and then select "Map resources" from the **Action** dropdown menu.
```
Should this be a list of steps, according to our UI guidelines? That will be easier for the AI bot to display.

{% endnavtab %}
{% navtab "API" %}

> 1. Create a service by sending a POST request to the [`/catalog-services` endpoint](/api/konnect/service-catalog/v1/#/operations/create-catalog-service):

Reviewer: ```suggestion
1. Create a Service Catalog service by sending a POST request to the [`/catalog-services` endpoint](/api/konnect/service-catalog/v1/#/operations/create-catalog-service):
```
At least in the first step.

<!--vale off-->
{% capture service %}
{% konnect_api_request %}
url: /v1/catalog-services
method: POST
status_code: 201
region: us
body:
  name: billing
  display_name: Billing Service
{% endkonnect_api_request %}
{% endcapture %}

{{ service | indent: 3 }}
<!--vale on-->

> 1. Map an integration resource to the service by sending a POST request to the [`/resource-mappings` endpoint](/api/konnect/service-catalog/v1/#/operations/create-resource-mapping):

Reviewer 2: See comment above.

> 1. Map an integration resource to the service by sending a POST request to the [`/resource-mappings` endpoint](/api/konnect/service-catalog/v1/#/operations/create-resource-mapping):

Author: ```suggestion
1. Map a resource to the service by sending a POST request to the [`/resource-mappings` endpoint](/api/konnect/service-catalog/v1/#/operations/create-resource-mapping):
```

---
title: "Service Catalog services"
content_type: reference
layout: reference

products:
    - service-catalog
works_on:
  - konnect

description: Learn about services in Service Catalog and how to configure them.

breadcrumbs:
  - /service-catalog/

related_resources:
  - text: "Service Catalog"
    url: /service-catalog/
  - text: Scorecards
    url: /service-catalog/scorecards/
  - text: Traceable integration
    url: /service-catalog/integrations/traceable/
  - text: GitHub integration
    url: /service-catalog/integrations/github/
  - text: GitLab integration
    url: /service-catalog/integrations/gitlab/
  - text: SwaggerHub integration
    url: /service-catalog/integrations/swaggerhub/
  - text: Datadog integration
    url: /service-catalog/integrations/datadog/
  - text: PagerDuty integration
    url: /service-catalog/integrations/pagerduty/
faqs:
  - q: What's the different between a Gateway Service and a Service Catalog service?
    a: |

>       A [Gateway Service](/gateway/entities/service/) represent the upstream services in your system and is the business logic component of your system that's responsible for responding to requests. A Service Catalog service is a collection of one or more resources from Service Catalog integrations.

Reviewer: ```suggestion
      A [Gateway Service](/gateway/entities/service/) is a {{site.base_gateway}} entity that represents an upstream service in your system and is the business logic component that's responsible for responding to requests. A Service Catalog service is a collection of one or more resources from Service Catalog integrations.
```
